### PR TITLE
docs: update and format changelog

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

This pull request updates and formats the changelog, prepping for the v5.0 release.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] confirmed that this PR does not change production code

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -65,30 +76,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-- [#1337](https://github.com/regen-network/regen-ledger/pull/1337) The `AddCreditType` proposal handler has been removed.
-- [#1342](https://github.com/regen-network/regen-ledger/pull/1342) The `NewKeeper` method in `ecocredit/marketplace` requires an `authority` address.
-- [#1342](https://github.com/regen-network/regen-ledger/pull/1342) The `AllowedDenom` proposal handler has been removed.
-- [#1357](https://github.com/regen-network/regen-ledger/pull/1357) The `NewModule` method in `ecocredit/module` requires a store `key`.
-- [#1462](https://github.com/regen-network/regen-ledger/pull/1462) Bridge operations must now specify a target/source that exists in the `AllowedBridgeChain` table.
-- [#1476](https://github.com/regen-network/regen-ledger/pull/1476) `DefaultParams` function has been removed.
-- [#1476](https://github.com/regen-network/regen-ledger/pull/1476) Remove `params` argument from `ValidateGenesis`.
-- [#1543](https://github.com/regen-network/regen-ledger/pull/1543) Remove `paramsKeeper` argument from `base/Keeper`, `basket/Keeper` and `marketplace/Keeper`.
```
Comment (ryanchristo) [inline CHANGELOG.md:83]: These items have been moved to their respective sections. I think we should be following the format we link to at the top of the document (at least for now) and this is an incomplete list at the moment.

Comment (ryanchristo) [reply CHANGELOG.md:83]: The migration guide is where we want to outline breaking changes for application developers. We also did a better job of using semantic commits this round but not quite ready to use as for auto-generating a changelog, which I still think is what we should continue working towards, which would have the breaking change indicator (i.e. `!`).

Comment (ryanchristo) [inline CHANGELOG.md:83]: These items have been moved to their respective sections. I think we should be following the format we link to at the top of the document (at least for now) and this was misleading as it is incomplete.

Comment (ryanchristo) [reply CHANGELOG.md:83]: The migration guide is where we outline breaking changes for developers consuming the API. I also think we are the only developers importing the repository at the moment and using the golang API but breaking changes in the golang should be included in the `Changes` section and/or other sections where appropriate.

Comment (ryanchristo) [reply CHANGELOG.md:83]: We did a better job of using semantic commits this round but not quite ready to use for auto-generating a changelog, which I still think is what we should continue working towards, which would have the breaking change indicator (i.e. `!`).

Comment (ryanchristo) [reply CHANGELOG.md:83]: We did a better job of using semantic commits this round but not quite ready to use for auto-generating a changelog, which is what we should continue working towards and would have the breaking change indicator (i.e. `!`).

**Review by clevinson — APPROVED** [review]

**Pull request merged**